merge nitropad and nitropc

alexgithublab opened the pull request.

daringer commented:
Looks pretty good for me, some comments:

* make sure the [syntax issues are resolved](https://github.com/Nitrokey/nitrokey-documentation/actions/runs/13495275806/job/37701043017?pr=402)
* remove debian, we don't provide this as a ISO anymore

jans23 commented on source/components/nitropad-nitropc/heads/index.rst:
```diff
@@ -0,0 +1,16 @@
+   Factory Reset Heads 2.0 <factory-reset-heads2>
+   Firmware Update <firmware-update>
+   Firmware Update v1.4+ <firmware-update-1.4>
+   System update <system-update>
```
"System Update"

jans23 commented on source/components/nitropad-nitropc/index.rst:
```diff
@@ -0,0 +1,18 @@
+   :maxdepth: 1
+   :glob:
+
+   QubesOS <qubes/index>
```
This page may confuse users because it doesn't say where to start. Clarify where to get started as in other products.

jans23 replied:
I don't think users should start with FAQ but with Getting Started in Ubuntu resp. Qubes.

jans23 replied:
This is still valid: This page may confuse users because it doesn't say where to start. Clarify where to get started as in other products.

jans23 commented on source/components/nitropad-nitropc/ubuntu/index.rst:
```diff
@@ -46,10 +58,29 @@ changed by you:
 .. include:: ../content/shared-index-content2.rst
+
+
+NitroPC
```
The headlines indicate that this document is split into NitroPad and NitroPC but instead the NitroPad section is partially relevant for NitroPC too.

jans23 replied:
same

alexgithublab commented:
bash syntax errors do not show with rstcheck_core 1.2.1

alexgithublab commented on source/components/nitropad-nitropc/qubes/index.rst:
```diff
@@ -37,4 +51,32 @@ changed by you:
 .. include:: ../content/shared-index-content2.rst
+
+
+NitroPC
```
NitroPC and NitroPad sections are removed

jans23 reviewed (changes requested):
- Remove `_ext` folder.
- The PR lacks redirects of the removed files/URLs.

alexgithublab commented:
Do you know a tool which I can use to retrieve the broken links ?

alexgithublab commented:
If I remove nitrokey-documentation/source/_ext/

then I cannot build a preview locally, here is the error I get:

Extension error:
Could not import extension nk_product_table (exception: No module named 'nk_product_table')
make: *** [Makefile:11: docs] Error 2

jans23 commented:
> Do you know a tool which I can use to retrieve the broken links ?

No recommendation but essentially it is all files you renamed or moved in this PR.

jans23 commented:
We can never fix the Internet and external sites and bookmarks. This is why whenever a file is removed or renamed, a redirect should be added.

alexgithublab commented:
I checked and I did not find any broken link related to Nitropad and NitroPC new configuration path

daringer commented:
`_ext` directory: this is needed in general, but your PR includes `__pycache__` from this directory, this shouldn't be the case, so please do not add any contents from within `_ext` into this PR 

redirects: we will need a redirect directive from the old structure to the new one:
* currently (w/o this PR) you can visit: https://docs.nitrokey.com/nitropad/qubes/firmware-update which will work as expected
* once this PR is merged this URL isn't working anymore, because it's now located here: https://docs.nitrokey.com/nitropad-nitropc/qubes/firmware-update
* so we have introduced "redirects" you can find here: https://github.com/Nitrokey/nitrokey-documentation/blob/main/redirects/.htaccess
* in this particular case a generic redirect should catch all URLs and forward them to the appropriate new URL in this PR, I think it should be something like that:
```
# from: https://docs.nitrokey.com/nitropad/qubes/firmware-update
# to: https://docs.nitrokey.com/nitropad-nitropc/qubes/firmware-update

RedirectMatch 301 "^(/[a-z][a-z])?/nitropad/(.*)$" "$1/nitropad-nitropc/$2"
RedirectMatch 301 "^(/[a-z][a-z])?/nitropc/(.*)$" "$1/nitropad-nitropc/$2"
```

please add these two lines to the `.htaccess` file and then using an "old" url should automatically redirect you to the "new" url ... (might also work for the preview - but cannot remember)

jans23 commented:
@alexgithublab Add such description about redirects to the README, please.

jans23 commented:
@alexgithublab Are you sure that no rst-file within "nitropc" and "nitropad" were  renamed or moved other than to "nitropad-nitropc"?

alexgithublab commented:
I did not renamed any files but yes I moved NitroPC folder content to nitropad-nitropc and also created the Heads folder and moved some files here as well.

jans23 commented:
You mean you created a subfolder "heads" which also changed the URL eventually? In this case you need to add a redirect for this too.

jans23 commented:
@daringer please have a look at the redirects.

daringer commented:
looks like this might work like this - we'll have to test this in detail once merged - will wait until tomorrow for that - please resolve the requested change until then

alexgithublab commented:
requested changes are not done yet ?
